src/crud.js

```diff
@@ -0,0 +1,134 @@
+                <i class="${editCheck}" id = "edit${i}"></i>`;
+    parent.appendChild(item);
+  }
+  /// remove buttons
+  const rmvBtnList = document.querySelectorAll('.trash');
+  rmvBtnList.forEach((element) => {
+    element.addEventListener('click', () => {
+      arr.splice(element.dataset.id, 1);
+      reOrder(arr);
+      localStorage.setItem('Task List', JSON.stringify(arr));
+      display(arr, parent);
+    });
+  });
+
+  /// /   update task descriptions function
+  function updateTask(arr, parent, id, positionID) {
+    const updatedText = document.getElementById(id).value;
+    if (updatedText !== '') {
+      arr[positionID].description = updatedText;
+      display(arr, parent);
+    }
+  }
+
+  /// /  editing action event listener
+  const editable = document.querySelectorAll('.edit');
+  editable.forEach((element) => {
+    element.addEventListener('click', () => {
+      const itemId = (element.id).charAt((element.id).length - 1);
+      for (let i = 0; i < arr.length; i += 1) {
+        if (i !== itemId) {
+          const editIcon = document.getElementById(`edit${i}`);
+          const input = document.getElementById(`input${i}`);
+          const trash = document.getElementById(`trash${i}`);
+          const item = document.getElementById(`item${i}`);
+          editIcon.classList.remove('rotate');
+          trash.classList.remove('visible');
+          item.classList.remove('bg-active');
+          input.classList.remove('focused');
+          input.readOnly = true;
+        }
+      }
+      element.classList.toggle('rotate');
+      const input = document.getElementById(`input${itemId}`);
+      const trash = document.getElementById(`trash${itemId}`);
+      const item = document.getElementById(`item${itemId}`);
+      trash.classList.toggle('visible');
+      item.classList.toggle('bg-active');
+      input.readOnly = false;
+      input.classList.toggle('focused');
+      input.addEventListener('keydown', (pressed) => {
+        if (pressed.key === 'Enter') {
+          updateTask(arr, parent, `input${itemId}`, itemId);
+          localStorage.setItem('Task List', JSON.stringify(arr));
+          display(arr, parent);
+        }
+      });
+    });
+  });
+
+  /// // check box event listener
+  const checkboxList = document.querySelectorAll('.checkbox');
+  checkboxList.forEach((item) => {
+    item.addEventListener('click', () => {
+      const itemId = (item.id).charAt((item.id).length - 1);
+      checkTask(itemId, arr, parent);
+      store(arr);
+      display(arr, parent);
+    });
+  });
+}
+
+/// //////   addTask function
+export function addTask(arr, id, parent) {
+  if (document.getElementById(id).value !== '') {
+    const record = {
+      description: document.getElementById(id).value,
+      completed: false,
+      index: arr.length + 1,
+    };
+    arr.push(record);
+    localStorage.setItem('Task List', JSON.stringify(arr));
```

Comment:
It would be nice if you have functions to get and set local storage data. This will make your code DRY.

function setLocalStorage = (array) => {
  localStorage.setItem('Task List', JSON.stringify(array));
}